docs/DeveloperGuide.md

```diff
@@ -1,34 +1,1080 @@
+
+----
+
+### 2.2 Before writing code
+
+#### 2.2.1 Configuring the Coding Style
+
+If using IDEA, follow this guide [IDEA: Configuring the code style](https://se-education.org/guides/tutorials/intellijCodeStyle.html) to setup IDEA’s coding style to match ours.
+    
+#### 2.2.2 Set up CI
+    
+There is no set up required as the project comes with a GitHub Actions config files, located in `.github/workflows` folder. When GitHub detects these files, it will run the CI for the project automatically at each push to the master branch or to any PR.
+    
+#### 2.2.3 Learn the Design
+    
+When you are ready to start writing codes, we recommended that you have a look at AniChan's overall design by reading about it at [AniChan's architecture](DeveloperGuide.md#31-architecture).
+    
+<br/>
+
+## 3. Design 
+
+The following section describes the architecture design of **Anichan**. This section starts off by looking at the overall architecture design in a general view, before going into the specific implementation details of the individual features.
+
+<br/>
+
+### 3.1 Architecture
+This section will help provide insight to the general overview of Anichan’s architecture.
+
+<br/>
+
+![Architectural Diagram](images/Architectural-Design.png)
+
+*Figure 1: Architecture Design Diagram*
+
+> :bulb: The images used are stored in the directory: `images/`. If you wish to update a diagram you may replace the images in this folder.
+
+<br/>
+
+The **Architecture Diagram** presented above explains the high-level design of AniChan, and given below is a quick overview of each component involved.
+
+The `Main` class is the starting point of the application and has only one class called `Main`, it is responsible for, 
+*   At launch: Initializes the various components in the correct sequence, connects them up with each other, and loads any saved data.
+*   At shut down: Shuts down the components and invokes any clean up methods where necessary.
+
+The rest of AniChan consists of 6 components: 
+*   `Ui`: Manages the user interface of AniChan.
+*   `Parser`: Parses the user input.
+*   `Command`: Executes the command.
+*   `User`: Manages the workspace(s) and user data.
+*   `AnimeData`: Provides data from the anime source file.
+*   `Storage`: Reads data from, and writes data to, the hard disk.
+
+<br/>
+
+Below is an overall sequence diagram to help illustrate the general program flow and how the different objects interact with each other.
+
+<br/>
+
+![Main Sequence Diagram](images/Overall-Sequence-Diagram.png) <br/>
+*Figure 2: Overall Sequence Diagram*
+ 
+<br/>
+
+### 3.2 UI Component
+![UI Class Diagram](images/Ui-Class-Diagram.png) <br/>
+*Figure 3: UI Class Diagram*
+
+The UI component consists of a `UI` class that handles all user input and system output. The UI is only dependent on the `Main` class and does not interact directly with other classes ensuring high cohesiveness and separation of roles.
```

Comment:
![image](https://user-images.githubusercontent.com/60333990/97656251-f658be80-1aa1-11eb-8bbe-a8a6cb6f9275.png)
It might be better UI could use the `UI` markdown as well to maintain the consistency of the markdowns.